apps/common/lib/lexical/code_unit.ex

```diff
@@ -180,21 +180,25 @@ defmodule Lexical.CodeUnit do
   end
 
   defp do_to_utf8(<<c::utf8, rest::binary>>, utf16_unit, utf8_unit) do
-    utf8_code_units = codepoint_size(c, :utf8)
-    utf16_code_units = codepoint_size(c, :utf16)
+    utf8_code_units = code_unit_size(c, :utf8)
+    utf16_code_units = code_unit_size(c, :utf16)
 
     do_to_utf8(rest, utf16_unit - utf16_code_units, utf8_unit + utf8_code_units)
   end
 
   @unicode_range 0..0x10_FFFF
 
-  defp codepoint_size(c, :utf16) when c in @unicode_range do
-    if c in 0x0000..0xD7FF or c in 0xE000..0xFFFF,
-      do: 1,
-      else: 2
+  defp code_unit_size(c, :utf16) when c in @unicode_range do
+    case c do
```

Comment:
So is this the only place where the logic was changed and else just was refactor?

Reply:
yes, the other logic was about codepoints, not code units.